### PR TITLE
Handle DataCenterMigration exceptions in all methods exposed by TelegramClient

### PR DESCRIPTION
**Pull request by Grepsy**
This handles the data center migration in more cases.

Fixes bugs like mentioned in #599.

### CODE REVIEW

**Comment by knocte** [comment]
Thanks for the PR @Grepsy! Can you reword the commit message to be exactly the same as the text in the PR please? This way, it will reference issue #599 automatically when merging it.

**Comment by Grepsy** [comment]
Done.

File: TLSharp.Core/TelegramClient.cs [hunk 1]
```diff
@@ -187,12 +187,20 @@ public bool IsUserAuthorized()
...
             return ((TLUser)request.Response.user);
         }
+        
         public async Task<TLPassword> GetPasswordSetting()
         {
             var request = new TLRequestGetPassword();
 
-            await _sender.Send(request);
-            await _sender.Receive(request);
+            try
```
Comment (knocte) [inline TLSharp.Core/TelegramClient.cs]: actually these try-catches you added are not correct; they need a while loop to work; please see how it's done in other similar try-catch blocks.

Comment (knocte) [reply TLSharp.Core/TelegramClient.cs]: No worries, can you now squash your two commits into one please? Easiest way to do it is using `git rebase -i HEAD~2`

**Comment by Grepsy** [comment]
Ok, I think this should be correctly squashed now.

**Comment by knocte** [comment]
Good stuff thanks!

**Pull request merged**